### PR TITLE
DDPPO Objectnav Agent and pretrained model

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
Provide agent wrapper, pre-trained model and submit instructions for [DDPPO Objectnav baseline](https://github.com/facebookresearch/habitat-api/pull/370).

- [x] Added ObjectNav DDPPO support to DDPPO agents.
- [x] Created ObjectNav DDPPO Docker file.
- [x] Added pre-trained ObjectNav DDPPO RGBD agent with 200 million steps.
- [x] Made Agent wrapper general to discrete action space.
- [x] Added instruction about ObjectNav baseline training and submission.

## How Has This Been Tested
Objectnav local test:
`docker build . --file Objectnav_DDPPO_baseline.Dockerfile -t objectnav_submission;bash test_locally_objectnav_rgbd.sh`

 Pointnav local test:
`docker build . --file Pointnav.Dockerfile -t pointnav_submission; ./test_locally_pointnav_rgbd.sh --docker-name pointnav_submission`

### CODE REVIEW

**Comment by dhruvbatra** [comment]
I was unable to test creating the DDPPO_Objectnav docker because my GCP VM kept going unresponsive every time I would try to. I suspect it's either filling up memory or space. But no point keeping this PR waiting.

**Comment by dhruvbatra** [comment]
@Skylion007 -- the DD-PPO PointNav docker creations is failing: 

```bash
(habitat) dbatra@ubuntu-bionic-1:~/habitat-challenge$ ls
CODE_OF_CONDUCT.md                   configs
CONTRIBUTING.md                      ddppo_agents.py
LICENSE                              ddppo_objectnav_habitat2020_challenge_baseline_v1.pth
Objectnav.Dockerfile                 habitat-challenge-data
Objectnav_DDPPO_baseline.Dockerfile  res
Pointnav.Dockerfile                  submission.sh
Pointnav_DDPPO_baseline.Dockerfile   test_locally_objectnav_rgbd.sh
README.md                            test_locally_pointnav_rgbd.sh
agent.py

(habitat) dbatra@ubuntu-bionic-1:~/habitat-challenge$ docker build . --file Pointnav_DDPPO_baseline.Dockerfile -t pointnav_submission
Sending build context to Docker daemon  98.69MB
Step 1/11 : FROM fairembodied/habitat-challenge:testing_2020_habitat_base_docker
 ---> bf52971c4a7f
Step 2/11 : RUN /bin/bash -c ". activate habitat; pip install ifcfg tensorboard && pip install torch==1.5.0+cu101 torchvision==0.6.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html"
 ---> Using cache
 ---> afbd1151493e
Step 3/11 : RUN /bin/bash -c ". activate habitat; git clone http://github.com/facebookresearch/habitat-api.git habitat-api2 && (cd habitat-api2 && git checkout 959bd45431edd8024832a877bdc8218015d97a7e) && cp -r habitat-api2/habitat_baselines habitat-api/."
 ---> Using cache
 ---> 20549b03767e
Step 4/11 : ADD ddppo_agents.py agent.py
 ---> Using cache
 ---> 70356dcb1fc5
Step 5/11 : ADD submission.sh submission.sh
 ---> Using cache
 ---> e854442eb860
Step 6/11 : ADD configs/challenge_pointnav2020.local.rgbd.yaml /challenge_pointnav2020.local.rgbd.yaml
 ---> Using cache
 ---> 70bd0a1adcd2
Step 7/11 : ADD configs/ configs/
 ---> Using cache
 ---> 4e2a35aa0b2f
Step 8/11 : ADD ddppo_pointnav_habitat2020_challenge_baseline_v1.pth demo.ckpt.pth
ADD failed: stat /var/lib/docker/tmp/docker-builder630335240/ddppo_pointnav_habitat2020_challenge_baseline_v1.pth: no such file or directory
```

**Comment by mathfac** [comment]
@dhruvbatra you missed `wget https://dl.fbaipublicfiles.com/habitat/data/baselines/v1/ddppo_pointnav_habitat2020_challenge_baseline_v1.pth`, double checking if it's still there.

File: README.md [hunk 1]
```diff
@@ -284,6 +193,94 @@ The challenge consists of the following phases:
...
+
+1. An example on how to train the DDPPO can be found in [habitat-api/habitat_baselines/rl/ddppo](https://github.com/facebookresearch/habitat-api/tree/master/habitat_baselines/rl/ddppo). See the corresponding README in habitat-api for how to adjust the various hyperparameters, save locations, visual encoders and other features. 
+    
+    1. To run on a single machine use the following script, where `$task={pointnav, objectnav}`:
```
[the remark this answers is not in the document]
Comment (mathfac) [reply README.md:209]: @dhruvbatra thanks, from "from `habitat-api/`" actually

File: README.md [hunk 2]
```diff
@@ -284,6 +193,94 @@ The challenge consists of the following phases:
...
+    
+    1. Finally modify the submission.sh script to run the appropiate command to test your agents. The scaffold for this code can be found ```agent.py``` and the DDPPO specific agent can be found in ```ddppo_agents.py```. In this example, we only modify the final command of the PointNav/ObjectNav docker: by adding the following args to submission.sh ```--model-path demo.ckpt.pth --input-type rgbd```. The default submission.sh script will pass these args to the python script. You may also replace the submission.sh. 
+        1. Please note that at this time, that habitat_baselines uses a slightly different config system and the configs nodes for habitat are defined under TASK_CONFIG which is loaded at runtime from BASE_TASK_CONFIG_PATH. We manually overwrite this config using the opts args in our agent.py file.
+
+1. Once your Dockerfile and other code is modified to your satisfcation, build it with the following command.
+    ```bash
+    docker build . --file ${Task}_DDPPO_Baseline.Dockerfile -t ${task}_submission
```
[the remark this answers is not in the document]
Comment (mathfac) [reply README.md:280]: Good catch!

**Pull request merged**